mail_edit/static/src/js/mail_edit.js

```diff
@@ -0,0 +1,73 @@
+/* © 2014-2017 Sunflower IT <www.sunflowerweb.nl>*/
+
+"use strict";
+openerp.mail_edit = function (instance) {
+    var _t = instance.web._t;
+    instance.mail.ThreadMessage.include({
+        bind_events: function () {
+            this._super.apply(this, arguments);
+            this.$('.oe_edit').on('click', this.on_message_edit);
+        },
+
+        on_message_edit: function () {
+            // Generate email subject as possible from record_name and subject
+            var subject = [];
+            if (this.record_name && (this.show_record_name ||
+                                     this.parent_id))
+            {
+                subject.push(this.record_name);
+            }
+            if (this.subject) {
+                subject.push(this.subject);
+            } else if (subject.length < 2) {
+                subject.push(_t("(No subject)"));
+            }
+
+            // Get only ID from the attachments
+            var attachment_ids = [];
+            for (var n in this.attachment_ids) {
```

Comment:
you're looking for http://underscorejs.org/#map

most of the code here can be a bit more elegant by using underscore/jquery stuff, but it takes some time to get a knack for this kind of programming. It's an odoo standard library, so I'd suggest to keep most of it (this one really should change, because the `in` operator in javascript doesn't always give you what you expect) but look into it for the future